pytensor/tensor/math.py

```diff
@@ -1571,6 +1574,63 @@ def std(input, axis=None, ddof=0, keepdims=False, corrected=False):
     return ret
 
 
+def median(input, axis=None):
+    """
+    Computes the median along the given axis(es) of a tensor `input`.
+
+    Parameters
+    ----------
+    input: TensorVariable
+        The input tensor.
+    axis: None or int or (list of int) (see `Sum`)
+        Compute the median along this axis of the tensor.
+        None means all axes (like numpy).
+    """
+    from pytensor.ifelse import ifelse
+
+    input = as_tensor_variable(input)
+    input_ndim = input.type.ndim
+    if axis is None:
+        axis = list(range(input_ndim))
+    elif isinstance(axis, int | np.integer):
+        axis = [axis]
+    elif isinstance(axis, np.ndarray) and axis.ndim == 0:
+        axis = [int(axis)]
+    else:
+        axis = [int(a) for a in axis]
```

Comment:
Can we use normalize_axis_tuple for these 3 cases like we do elsewhere?

Reply:
Surely. It won't work in cases of ```axis=None``` right?
Stuff like ```var``` and ```mean``` use this so should we open an issue to replace those?

Reply:
Yeah, we should use it everywhere, and no I don't think it handles axis=None, but we can double check